### ISSUE TITLE
Add support for Diff mode in JUnit 5

### ISSUE THREAD

**Issue by lordofthejars** [SpectoLabs/hoverfly-java#206]
Add support for Diff mode in JUnit 5 so you can use annotations instead of a programmatic way.

This implies creating two annotations `@HoverflyDiff` (`Hoverfly hoverfly = new Hoverfly(configs(), DIFF) hoverfly.importSimulation(classpath("simulation-to-compare.json"));`) 

and `@HoverflyValidate` ( hoverfly.assertThatNoDiffIsReported(false);)

I am starting working on it

**Comment by tommysitu** [comment]
That'll be great! 

Also worth to mention that you can filter out headers when doing a diff since hoverfly [v0.17.3](https://github.com/SpectoLabs/hoverfly/releases/tag/v0.17.3)

Presumably, `@HoverflyValidate` can be used on both class and method levels?

What about calling it `@HoverflyCompare` because Diff is for comparing the simulation and the real response? What do you think @JohnFDavenport?

**Comment by lordofthejars** [comment]
@tommysitu is this filter header feature implemented in Hoverfly Java too?

**Comment by tommysitu** [comment]
It uses the same flags as capture mode, so you can kind of enable it using the `configs.captureHeaders(...)` but obviously it looks very strange. I will create an issue for improving this feature and documentation.

**Comment by lordofthejars** [comment]
but with captureHeaders you set the headers you want to use in Diff or the ones you don't want to use?

**Comment by tommysitu** [comment]
The ones that you want to filter out in Diff mode.

**Comment by lordofthejars** [comment]
But it is called `capture` but you are applying as filter out. IIUC `filter out` it seems that is not the same right?

**Comment by tommysitu** [comment]
Diff mode happens to use the same mode argument flag `-headers` to do header filtering. For setting the flag in Hoverfly-Java, you can only use `captureHeaders` config options at the moment which doesn't make sense for the diff mode. That's why I think a new config option should be defined for diff mode to do that, although it passes the same flag under the hood. Don't worry, this will be a separate issue.

**Comment by lordofthejars** [comment]
I don't know if there is a bug but I am using `.captureHeaders("Diferencia")` to just take into consideration this field and it is not working. 

Then what I do is `.captureHeaders("Date")` which is the one I would like to not be used in diff.

But in both cases I get:

```
1. diff report at 2019-03-16T12:08:25+01:00:
(1.) The "header/Date" parameter is not same - the expected value was [[Fri, 15 Mar 2019 18:22:53 GMT]], but the actual one [[Sat, 16 Mar 2019 11:08:25 GMT]]
```
So I am not sure if I am using the API in a wrong way or there is a bug somewhere.

**Comment by lordofthejars** [comment]
Ok the bug is in `Hoverfly.java` line 517 `setModeWithArguments` that is only setting the capture Headers in case of CAPTURE mode and not DIFF. I'll add an if mode == DIFF and headers not empty then set both.

**Comment by tommysitu** [comment]
Thank you!

**Comment by lordofthejars** [comment]
PR done: https://github.com/SpectoLabs/hoverfly-java/pull/207

**Comment by tommysitu** [comment]
This feature will be available in v0.11.3.
